test/functional/bind.cpp

```diff
@@ -0,0 +1,387 @@
+/*
+  SLB.Functional
+
+  Copyright Michael Park, 2017
+  Copyright Agustin Berge, 2017
+
+  Distributed under the Boost Software License, Version 1.0.
+  (See accompanying file LICENSE.md or copy at http://boost.org/LICENSE_1_0.txt)
+*/
+
+#include <slb/functional.hpp>
+
+#include <functional>
+#include <type_traits>
+#include <utility>
+
+#include "../catch.hpp"
+
+#define CHECK_NESTED(...)                                                      \
+  do {                                                                         \
+    INFO(__FILE__ "(" << __LINE__ << "): " #__VA_ARGS__);                      \
+    check_##__VA_ARGS__;                                                       \
+  } while (false)
+
+// [func.bind], bind
+
+// template<class T> struct is_bind_expression;
+TEST_CASE("is_bind_expression", "[func.bind.isbind]") {
+  struct S {
+    static int f(int base) { return base; }
+  };
+  using bind_t = decltype(std::bind(&S::f, 42));
```

Comment:
Is this intentionally testing for `slb::is_bind_expression<decltype(std::bind(&S::f, 42))>`?
Seems like it should be `slb::bind` instead, this seems more like a `std-compatible` test.

Reply:
Yes, this is intentional, given that `is_bind_expression` is what's being tested. I'm going to add `slb` counterparts just as I do for `is_placeholder`.